### DISCUSSION TITLE
How to get results of expect_exit?

### DISCUSSION THREAD

**Discussion by LLsmile** [google/googletest#4085] [Community Help]
I want to do a test in a loop like this:
**for(int i = 0; i < 1000; i++) {
    for(int j = 0; j < 1000; j++) {
        auto &result = EXPECT_EXIT(XXXXXXXXXXX);
        if (result) { 
            do something;
            break; // jump out of loop for j 
        }
    }
}**
Now, what I need is the result of the `EXPECT_EXIT`.

**Comment by LLsmile** [comment]
In my test case, I want to test about 1000 functions with different inputs. So I need to test all 1000 functions one by one. For each function, I want to test with different inputs. But once it failed, this function should be record and stopped.

**Comment by derekmauro** [comment]
If I understand correctly, you are trying to get the values of `i` and `j` when the test fails?

If so, you can use `<<` to log extra information when the test fails. You can also use `ASSERT_*` instead of `EXPECT_*` to stop immediately as well.

```
TEST(DeathTest, Test) {
  for (int i = 0; i < 10; ++i) {
    ASSERT_EXIT(std::exit(i), testing::ExitedWithCode(i), "") << "i=" << i;
  }
}
```

It is also possible that I don't understand what you are trying to do. If so, please provide a full example.

**Reply by LLsmile** [reply]
Thanks for your reply. For example, I have 5 functions to be tested with different parameters. And some functions may crash in certain environment. So, this is my code:

std::vector<Function> functions;
for(int i= 0; i < 5; i++) {
  auto parameters = functions[i].get_parameter_candidates();
  for(const auto &p: parameters) {
    auto &results = EXPECT_EXIT( try { functions[i].test(p); exit(0);} catch(...) {exit(0);}, ::testing::ExitedWithCode(0), "");
    if (results) {do something; break}
  }
}

**_EXPECT_EXIT_** records all crashed functions and ignore all exception. **But I want to know whether this functions crash to do something else.**
